Sources/Algorithms/LazySplitCollection.swift

```diff
@@ -0,0 +1,349 @@
+  /// around elements equal to the given element.
+  ///
+  /// The resulting lazy sequence consists of at most `maxSplits + 1` subsequences.
+  /// Elements that are used to split the collection are not returned as part of any
+  /// subsequence (except possibly the last one, in the case where `maxSplits` is
+  /// less than the number of separators in the collection).
+  ///
+  /// The following examples show the effects of the `maxSplits` and
+  /// `omittingEmptySubsequences` parameters when splitting a string at each
+  /// space character (" "). The first use of `split` returns each word that
+  /// was originally separated by one or more spaces.
+  ///
+  ///     let line = "BLANCHE:   I don't want realism. I want magic!"
+  ///     for spaceless in line.lazy.split(separator: " ") {
+  ///       print(spaceless)
+  ///     }
+  ///     // Prints
+  ///     // BLANCHE:
+  ///     // I
+  ///     // don't
+  ///     // want
+  ///     // realism.
+  ///     // I
+  ///     // want
+  ///     // magic!
+  ///
+  /// The second example passes `1` for the `maxSplits` parameter, so the
+  /// original string is split just once, into two new strings.
+  ///
+  ///     for spaceless in line.lazy.split(separator: " ", maxSplits: 1) {
+  ///       print(spaceless)
+  ///     }
+  ///     // Prints
+  ///     // BLANCHE:
+  ///     // I don't want realism. I want magic!
+  ///
+  /// The final example passes `false` for the `omittingEmptySubsequences`
+  /// parameter, so the returned array contains empty strings where spaces
+  /// were repeated.
+  ///
+  ///     for spaceless in line.lazy.split(separator: " ", omittingEmptySubsequences: false) {
+  ///       print(spaceless)
+  ///     }
+  ///     // Prints
+  ///     // BLANCHE:
+  ///     //
+  ///     //
+  ///     // I
+  ///     // don't
+  ///     // want
+  ///     // realism.
+  ///     // I
+  ///     // want
+  ///     // magic!
+  ///
+  /// - Parameters:
+  ///   - separator: The element that should be split upon.
+  ///   - maxSplits: The maximum number of times to split the collection, or
+  ///     one less than the number of subsequences to return. If
+  ///     `maxSplits + 1` subsequences are returned, the last one is a suffix
+  ///     of the original collection containing the remaining elements.
+  ///     `maxSplits` must be greater than or equal to zero. The default value
+  ///     is `Int.max`.
+  ///   - omittingEmptySubsequences: If `false`, an empty subsequence is
+  ///     returned in the result for each consecutive pair of `separator`
+  ///     elements in the collection and for each instance of `separator` at
+  ///     the start or end of the collection. If `true`, only nonempty
+  ///     subsequences are returned. The default value is `true`.
+  /// - Returns: A lazy sequence of subsequences, split from this collection's
+  ///   elements.
+  ///
+  /// - Complexity: O(*n*), where *n* is the length of the collection.
+  func split(
```

Comment:
Should this be `@inlinable`?
I think we should look to make functions `@inlinable` or `@usableFromInline` where it makes sense :)

Reply:
Definitely, inlinability is on my to-do list. Currently working on making `LazySplitCollection` conform to `Collection`.